### ISSUE TITLE
When testing on datasets containing only real images the performance is poor

### ISSUE THREAD

**Issue by SilverRAN** [chuangchuangtan/NPR-DeepfakeDetection#9]
Thanks for your wonderful work. It's a very simple but effective method when tesing on images generated by deep learning models. But I found that if I test the pretrained model with only real images, the accuracy is only around 30%. It seems the model prefer predicting "fake" rather than "real". Is this normal?

**Comment by chuangchuangtan** [comment]
I would appreciate it if you could provide me with the specific test dataset.

**Comment by SilverRAN** [comment]
> I would appreciate it if you could provide me with the specific test dataset.

Sure. The testing data can be downloaded from https://github.com/VL-Group/Natural-Color-Fool/releases/download/data/images.zip

**Comment by SilverRAN** [comment]
> I would appreciate it if you could provide me with the specific test dataset.

I tested the Huggingface demo and found its predictions are more accurate. Could you please provide the code for reading input in your Huggingface demo? Thanks a lot.

**Comment by chuangchuangtan** [comment]
Our released model did not consider disturbances such as JPEG compression and noise during training, hence it performs poorly on images in online social networks.  Increasing data augmentation during the training process can improve robustness.

You can directly access the inference code on Huggingface through the "Files" tab in the upper right corner of the demo.

**Comment by zj56** [comment]
def npr(x:torch.Tensor):
    B, C, H, W = x.size()
    assert H==W
    unfold = torch.nn.Unfold(kernel_size=(2, 2), stride=2)
    fold = torch.nn.Fold(output_size=(H,W), kernel_size=(2, 2), stride=2)
    x= unfold(x)  # shape = [*, width, grid ** 2]  grid=2
    x_ = x.clone()
    for i in range(0, 12, 4):
        diffenerce = x[:, i, :].unsqueeze(dim=1)
        x[:, i:i + 4, :] = x_[:, i:i + 4, :] - diffenerce
        # print(diffenerce)
    x= fold(x)
    return x
